### ISSUE TITLE
Accessibility improvements on multiple select dropdowns

### ISSUE THREAD

**Issue by mlinnetz** [Choices-js/Choices#1348]
**Describe the bug**
In Choices Multi-select inputs, after selecting an option, it adds it to a "Selected" container with **role="listbox"**. Each selected option then gets added as a `<div role="option">` with a Remove Item `<button /> `inside the div.
This is an accessibility and WCAG-compliance issue, because elements with **role="option"** cannot contain other elements, especially not actionable elements like buttons.   `<button>` is not an allowed descendant of an element with **role="listbox"** or **role="option"**. This confuses the assistive technologies.
See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/listbox_role for more information

**To Reproduce**
Implement any choices `<select multiple>` multi-select input, 

**Expected behavior**
### Proposed resolution

In the interest of getting closer to WCAG compliance, I would put forth a few proposed changes for choices multi-select dropdown, e.g.,` <select multiple>`.

1.  The selected options be placed in an regular unordered list ( `<ul><li>`) instead of the current
    `<div role="listbox"><div role="option">`.  You can add the Remove Item button to each `<li>`. There's no reason for the list of selected items to be a listbox, right?   The original choices list (choices__list) is still present and hidden.
2.  The choices combobox element (e.g., `<div role="combobox">`) should have **aria-haspopup="listbox"** and **aria-controls="xxxxx"** attributes indicating that it controls the choices select list (e.g., `<div class="choices__list" role="listbox">`). This would require assigning a unique id attribute to the choices select list so that the **aria-controls** attribute can target the list.
3. The choices select list (e.g., `<div class="choices__list" role="listbox">`) should have an **aria-labelledby** or **aria-label**. This improves screenreader functionality because when you open the select list, it announces "List for ...." instead of just "List". 
 I believe that is the intention of the labelId parameter? That labelId parameter should add the label to both the choices element (**role="combobox"**) as well as the select list.

My proposals above needs validation, for example we need to confirm whether it would break something else which was relying on the selected items to have that role of **listbox** or **option**.
Furthermore, it might be worth it to get an a11y expert to review/validate these changes and/or propose changes more holistically. 

**Choices version and bundle**
- Version: latest version, currently v.11.1.0

**Desktop (please complete the following information):**
All browsers

**Additional context**
